### PR TITLE
Basic utilities and extensions

### PR DESCRIPTION
**Pull request by muneshkumar578**
`Sockethead.ExtensionsAndUtilities` is a collection of utilities and extension methods that can be utilized in any project. It provides various useful utilities and extensions for string, JSON, collection, date, and more.

### CODE REVIEW

File: Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        /// <param name="startDate">Start date and first element of enumerable</param>
+        /// <param name="endDate">End date for enumerable</param>
+        /// <returns>Enumerable for the days in the range</returns>
+        public static IEnumerable<DateTime> GetDateRange(DateTime startDate, DateTime endDate)
+        {
+            for ( ; startDate <= endDate; startDate = startDate.AddDays(1))
+                yield return startDate;
+        }
+    }
+}
```
Comment (rjeike) [inline Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs:23]: You could make this an extension to DateTime (where startDate is this)

**Comment by rjeike** [comment]
It looks good, with one exception, the name.  Sockethead.ExtensionsAndUtilities – yuck!

How about Sockethead.Common?

Basically it is the place were we put any non-framework dependent things.  It aligns with the structure where we have Sockethead.Razor and Sockethead.EFCore which are framework dependent.

What do you think?

File: Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    public static class DateUtils
+    {
+        /// <summary>
+        /// Returns a range of DateTime that starts with the startDate and ends with the endDate, inclusive of both endpoints.
+        /// </summary>
+        /// <param name="startDate">Start date and first element of enumerable</param>
+        /// <param name="endDate">End date for enumerable</param>
+        /// <returns>Enumerable for the days in the range</returns>
+        public static IEnumerable<DateTime> GetDateRange(DateTime startDate, DateTime endDate)
+        {
```
Comment (rjeike) [inline Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs]: You could make this an extension to DateTime (where startDate is this)

**Comment by muneshkumar578** [comment]
> It looks good, with one exception, the name. Sockethead.ExtensionsAndUtilities – yuck!
> 
> How about Sockethead.Common?
> 
> Basically it is the place were we put any non-framework dependent things. It aligns with the structure where we have Sockethead.Razor and Sockethead.EFCore which are framework dependent.
> 
> What do you think?

Naming things is one of the hardest parts of computer science :) I'm glad we're on the same page about the name change. Thanks for the name suggestion. I also think that Sockethead.Common is a much more descriptive and concise term that effectively conveys the purpose of the package as a collection of non-framework dependent tools and utilities.

File: Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs (hunk 2 above)
Comment (muneshkumar578) [reply Sockethead.ExtensionsAndUtilities/Utilities/DateUtils.cs]: Moved this to the extension method.

**Review by rjeike — APPROVED** [review]

**Pull request merged**